File: Intterop-Excel-Word C#/Form1.cs
using System;
using System.Windows.Forms;
using System.IO;
using Excel = Microsoft.Office.Interop.Excel;
using Word = Microsoft.Office.Interop.Word;

namespace Intterop_Excel_Word
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }




        private void button1_Click(object sender, EventArgs e)
        {
            string Nombre = Tbx_Nombre.Text;
            string Texto = Tbx_Texto.Text;

            string path = Tbx_Directorio.Text + "\\" + Nombre + ".xlsx";
            if (File.Exists(path))
            {
                File.Delete(path);
            }

            Excel.Application oApp;
            Excel.Worksheet oSheet;
            Excel.Workbook oBook;
            Excel.Workbook wb;           

            oApp = new Excel.Application();
            oApp.Visible = true;
            oBook = oApp.Workbooks.Add();
            oSheet = (Excel.Worksheet)oBook.Worksheets.get_Item(1);
            oSheet.Cells[1, 1] = Texto;

            oBook.SaveAs(path);
            
        }

        private void Btn_Directorio_Click(object sender, EventArgs e)
          {
            FolderBrowserDialog Directorio = new FolderBrowserDialog();
            DialogResult = Directorio.ShowDialog();
            if(DialogResult == DialogResult.OK) 
              {
                Tbx_Directorio.Text = Directorio.SelectedPath;
              }
          }

        private void button2_Click(object sender, EventArgs e)
        {
            string Nombre = Tbx_Nombre.Text;
            string Texto = Tbx_Texto.Text;

            string path = Tbx_Directorio.Text + "\\" + Nombre;

            object ObjMiss = System.Reflection.Missing.Value;
            Word.Application oApp = new Word.Application();
            Word.Document oDocument = oApp.Documents.Add(ref ObjMiss, ref ObjMiss, ref ObjMiss, ref ObjMiss);
            oDocument.Activate();
            oApp.Selection.TypeText(Tbx_Texto.Text);
            oApp.Visible = true;
            oDocument.SaveAs2(path);
          } 
    }
}

File: Interop Excel-Word VB/Form1.vb
Imports Excel = Microsoft.Office.Interop.Excel
Imports Word = Microsoft.Office.Interop.Word

Public Class Form1
    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub button1_Click(sender As Object, e As EventArgs) Handles button1.Click
        Dim Nombre As String = Tbx_Nombre.Text
        Dim Texto As String = Tbx_Texto.Text

        Dim path As String = Tbx_Directorio.Text & "\" & Nombre & ".xlsx"

        Dim oApp As Excel.Application
        Dim oSheet As Excel.Worksheet
        Dim oBook As Excel.Workbook
        Dim wb As Excel.Workbook
        oApp = New Excel.Application()
        oApp.Visible = True
        oBook = oApp.Workbooks.Add()
        oSheet = CType(oBook.Worksheets.Item(1), Excel.Worksheet)
        oSheet.Cells(1, 1) = Texto
        oBook.SaveAs(path)
    End Sub

    Private Sub button2_Click(sender As Object, e As EventArgs) Handles button2.Click
        Dim Nombre As String = Tbx_Nombre.Text
        Dim Texto As String = Tbx_Texto.Text

        Dim path As String = Tbx_Directorio.Text & "\" & Nombre

        Dim ObjMiss As Object = System.Reflection.Missing.Value
        Dim oApp As Word.Application = New Word.Application()
        Dim oDocument As Word.Document = oApp.Documents.Add(ObjMiss, ObjMiss, ObjMiss, ObjMiss)
        oDocument.Activate()
        oApp.Selection.TypeText(Tbx_Texto.Text)
        oApp.Visible = True
        oDocument.SaveAs2(path)
    End Sub

    Private Sub Btn_Directorio_Click(sender As Object, e As EventArgs) Handles Btn_Directorio.Click
        Dim Directorio As FolderBrowserDialog = New FolderBrowserDialog()
        DialogResult = Directorio.ShowDialog()

        If DialogResult = DialogResult.OK Then
            Tbx_Directorio.Text = Directorio.SelectedPath
        End If
    End Sub
End Class
